swap/src/bin/swap.rs

```diff
@@ -94,23 +94,24 @@ async fn main() -> Result<()> {
                 EventLoop::new(swap_id, swarm, alice_peer_id, bitcoin_wallet.clone())?;
             let event_loop = tokio::spawn(event_loop.run());
 
-            let send_bitcoin = determine_btc_to_swap(
-                event_loop_handle.request_quote(),
-                bitcoin_wallet.balance(),
-                bitcoin_wallet.new_address(),
-                async {
-                    while bitcoin_wallet.balance().await? == Amount::ZERO {
-                        bitcoin_wallet.sync().await?;
+            debug!("Requesting quote");
+            let bid_quote = event_loop_handle.request_quote().await?;
+            info!("Received quote: 1 XMR ~ {}", bid_quote.price);
 
-                        tokio::time::sleep(Duration::from_secs(1)).await;
-                    }
+            let max_givable = || bitcoin_wallet.max_giveable(TxLock::script_size());
 
-                    bitcoin_wallet.balance().await
-                },
-                bitcoin_wallet.max_giveable(TxLock::script_size()),
+            let (send_bitcoin, fees) = determine_btc_to_swap(
+                bid_quote.clone(),
+                max_givable().await?,
+                bitcoin_wallet.new_address(),
+                || bitcoin_wallet.balance(),
+                max_givable,
```

Comment:
Just pass `max_givable` once.

Reply:
I guess you did that for the tests? You should be able to have state inbetween the calls to the Fn but might need to change the type to `FnMut`.

Reply:
Yes, I did it for the tests. I think it's fine to leave it with the initial value passed in. I feel otherwise the tests get unnecessary complex. I was trying to put as little "logic" into the tests as possible.